R/utility_spliceHelper_functions.R

```diff
@@ -1,598 +1,572 @@
+                                                       strand)
+  annotatedTable$internalFirstExon.query <- annotateInternalStartEnd(queryStartRng,
+                                                                    subjectFullRng, annotatedTable$alternativeFirstExon)
+  annotatedTable$internalFirstExon.subject <- annotateInternalStartEnd(subjectStartRng,
+                                                                      queryFullRng, annotatedTable$alternativeFirstExon)
+  # TES/last exon
+  annotatedTable$alternativeLastExon <- alternativeStartEndExon(queryEndRng, 
+                                                               subjectEndRng)
+  annotatedTable$alternativeTES <- calculateTESdistance(queryEndRng, 
+                                                       subjectEndRng, 
+                                                       annotatedTable$alternativeLastExon,
+                                                       strand)
+  annotatedTable$internalLastExon.query <- annotateInternalStartEnd(queryEndRng,
+                                                                   subjectFullRng, annotatedTable$alternativeLastExon)
+  annotatedTable$internalLastExon.subject <- annotateInternalStartEnd(subjectEndRng,
+                                                                     queryFullRng, annotatedTable$alternativeLastExon)
+  #intron retention subject and query
+  annotatedTable$intronRetention.subject <- annotateIntronRetent(querySpliceRng, subjectFullRng)
+  annotatedTable$intronRetention.query <- annotateIntronRetent(subjectSpliceRng ,queryFullRng)
+  #exon skipping query and subject
+  annotatedTable$exonSkipping.query <- annotateExonSkip(querySpliceRng, subjectFullRng,
+                                                        subjectStartRng, subjectEndRng)
+  annotatedTable$exonSkipping.subject <- annotateExonSkip(subjectSpliceRng ,queryFullRng,
+                                                          queryStartRng, queryEndRng) 
+  # exon 3' splice site
+  exonSpliceTable <- annotateExonSplice(querySpliceRng, subjectFullRng,
+                                        subjectStartRng, subjectEndRng,
+                                        annotatedTable$strand)
+  annotatedTable <- cbind(annotatedTable, exonSpliceTable)
+  return(annotatedTable)
+}
 
-  subjectEndExonStartExtension <- start(subjectEnd.querySplice.Rng)<end(qSpRngList) & end(subjectEnd.querySplice.Rng)>end(qSpRngList) & start(qSpRngList)< start(subjectEnd.querySplice.Rng)
-  subjectEndExonStartExtension <- tapply(subjectEndExonStartExtension, subjectStartEnd.querySplice.IdMap, sum)
+#' extract strand from GRangesList
+#' @noRd
+getStrandFromGrList <- function(grl) { 
+  return(unlist(strand(grl), use.names = F)[cumsum(elementNROWS(grl))]) 
+  }
+
+#' start/end ranges pre-processing                                 
+expandRanges <- function(unexpandedRng,targetRng){ 
```

Comment:
Here we might not need to add the full documentation, but can you add comments on what this function does?

Reply:
I have added a description to this function.